keepalived/check/check_daemon.c

```diff
@@ -253,7 +253,11 @@ reload_check_thread(__attribute__((unused)) thread_t * thread)
 	thread_cleanup_master(master);
 	free_global_data(global_data);
 
-	free_checkers_queue();
+	/* Save previous checker data */
+	old_checkers_queue = checkers_queue;
+	checkers_queue = NULL;
+	ncheckers = 0;
```

Comment:
Does not necessarily need to reset checker-id, but I think it prevent overflow.

I believe that even if the checker-id is duplicated before and after the reload, there is no problem. The old checker is discarded after the reloading process is over.